spras/containers.py

```diff
@@ -148,6 +150,42 @@ def run_container(framework: str, container_suffix: str, command: List[str], vol
     else:
         raise ValueError(f'{framework} is not a recognized container framework. Choose "docker" or "singularity".')
 
+def run_container_and_log(name: str, framework: str, container_suffix: str, command: List[str], volumes: List[Tuple[PurePath, PurePath]], working_dir: str, environment: str = 'SPRAS=True'):
+    """
+    Runs a command in the container using Singularity or Docker with associated pretty printed messages.
+    @param framework: singularity or docker
+    @param container_suffix: name of the DockerHub container without the 'docker://' prefix
+    @param command: command to run in the container
+    @param volumes: a list of volumes to mount where each item is a (source, destination) tuple
+    @param working_dir: the working directory in the container
+    @param environment: environment variables to set in the container
+    @return: output from Singularity execute or Docker run
+    """
+    print('Running {} on framework "{}" with command: {}'.format(name, framework, ' '.join(command)), flush=True)
```

Comment:
I've always thought the word "framework" is a strange way to expose the name of the container runtime to users. Personally, I'd either explicitly say "with container framework {}" or maybe "with container runtime {}". Of course, it's also nice to use consistent language for developers, and switching "framework" to "runtime" may confuse people.

Just my two cents on this one -- up to you whether you want to change anything!